[CORE-10012] rptests/tests: Fix consumer group stress test

IoannisRP opened the pull request:
Fixes: [CORE-10012](https://redpandadata.atlassian.net/browse/CORE-10012)

Sometimes, this test would take more than 10 minutes to run, which is the default value of `group_offset_retention_check_ms`. This would trigger the `group_offset_retention_sec`, which was set to only 20sec in this test. Then when the client tries to get the offsets, after the consumers have been stopped, it would return an error value (-1001).

To address missing offsets in such cases, offset retention period has been disabled.

The slowdown happens because some commit requests block for a long time when the leadership transfer happens. It appears that they block due to blocked fetch requests in the librdkafka request queue. When reducing the value of fetch.wait.max, the problem seems to go away.

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x

## Release Notes

* none



[CORE-10012]: https://redpandadata.atlassian.net/browse/CORE-10012?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

IoannisRP committed:
rptests/tests: Disable retention period in consumer group stress test

Sometimes, this test would take more than 10 minutes to run, which
is the default value of `group_offset_retention_check_ms`. This would
trigger the `group_offset_retention_sec`, which was set to only 20sec
in this test. Then when the client tries to get the offsets, after
the consumers have been stopped, it would return an error value (-1001).

As this test is purpose is correctness under heavy load, not performance,
offset retention is disabled.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--20528059-a1a6-41c9-ba07-c74418756c8a-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/68087">build#68087</a></summary>


| test_class               | test_method          | test_arguments                                                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                      |
|:-------------------------|:---------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------|
| AuditLogTestKafkaApi     | test_no_auth_enabled |                                                                                                                                                | ducktape    | https://buildkite.com/redpanda/redpanda/builds/68087#0197b265-3f12-4b95-b23d-909d3c9e4847 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS   |
| RandomNodeOperationsTest | test_node_operations | {"cloud_storage_type": 2, "compaction_mode": "chunked_sliding_window", "enable_failures": true, "mixed_versions": true, "with_iceberg": false} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/68087#0197b265-3f14-4f9b-819e-e26870586620 | FLAKY         | 17/21    | upstream reliability is '100.0'. current run reliability is '80.95238095238095'. drift is 19.04762 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->

dotnwat commented on tests/rptest/tests/consumer_group_test.py:
```diff
@@ -1604,7 +1604,7 @@ def __init__(self, test_context):
                                  "group_topic_partitions": 1,
                                  "compacted_log_segment_size": 1024 * 1024,
                                  "log_compaction_interval_ms": 1000,
-                                 "group_offset_retention_sec": 20,
+                                 "group_offset_retention_sec": None
```
but is this a realistic setting? maybe the test needs to be updated to be more tolerant of retention settings?

IoannisRP replied:
It doesn't look like performance is a consideration for this test. It looks to me that this tests verifies that even under heavy load, the results are the expected. Not that they are done on time.

Also, i would argue that the previous setting of `20s` doesn't make much sense when `group_offset_retention_check_ms` is left on its default value of `10min`.

@mmaslankaprv what do you think?

mmaslankaprv replied:
maybe let's set that value to a bit more than 20s, it shouldn't trigger as the committers should update the offsets, so the retention should not be applied when the offset was updated,

IoannisRP replied:
@mmaslankaprv
The time it takes for the
https://github.com/redpanda-data/redpanda/blob/5d19f25616902d6a1a684f985afd6ef78f4fc8f6/tests/rptest/tests/consumer_group_test.py#L1690-L1691
takes anywhere from 1min - 45mins on CI. These numbers are by looking through successful runs (in the range 1-5 mins) and failed runs ( >30 mins).

My understanding is that if it goes above the `group_offset_retention_check_ms` time (which is `10min`), while stopping consumers, it will fail, as the offsets will be not there. The timing in question is from stopping the first consumer to requesting the first consumer's offset. While theoretically it can happen for any consumer, this is the most probable to exhibit this behavior.

I am not sure how setting the `group_offset_retention_sec` ~1min or so will help with the flakiness.

mmaslankaprv replied:
that makes sense, i think we can disable retention in this test

dotnwat replied:
@IoannisRP please consider summarizing conversations like this and adding them either as a comment in the code or into the commit message. future people reading the test will undoubtedly wonder why retention is disabled and it's nice when github isn't in the path to discovering the answer.

pgellert reviewed (dismissed):
lgtm, but apart from this, I think we should also figure out why the test is taking so long to stop because tests hanging for >30min are as problematic as CI failures

IoannisRP committed:
rptests/tests: Reduce fetch max wait time for consumer group stress test

During the partition leadership transfers, some of the ongoing consumer
commit requests would block for a very long time (up to multiple
minutes). This seems to be due to fetch requests in the librdkafka
request queue, as reducing the fetch.wait.max from default value of
500ms to 50ms seems to completely solve the issue.

IoannisRP commented:
Changes in [force-push](https://github.com/redpanda-data/redpanda/compare/007938c91d6408b43a95db07f70e8c43f1e7dbe2..7f2b9555b5045410f01d121fb69b8e51d07b27a3):
- Reworded commit about offset retention
- Addressed issue for slowdown

dotnwat commented:
Thanks @IoannisRP for the updated commit message

IoannisRP commented:
/ci-repeat 1
dt-repeat=50
tests/rptest/tests/consumer_group_test.py::ConsumerGroupOffsetResetTest.test_stress_consumer_group_commits

IoannisRP commented:
/ci-repeat 1
skip-units
dt-repeat=50
tests/rptest/tests/consumer_group_test.py::ConsumerGroupOffsetResetTest.test_stress_consumer_group_commits

IoannisRP commented:
/ci-repeat 1
skip-units
dt-repeat=50
tests/rptest/tests/consumer_group_test.py::ConsumerGroupOffsetResetTest.test_stress_consumer_group_commits
